packages/scene-system/sdk/SceneRuntime.ts

```diff
@@ -144,36 +142,47 @@ export async function startSceneRuntime(client: RpcClient) {
     sendBatchAndProcessEvents().catch(devToolsAdapter.error).finally(reschedule)
   }
 
-  const dcl = createDecentralandInterface({
-    clientPort,
-    onError: (err: Error) => devToolsAdapter.error(err),
-    onLog: (...args: any) => devToolsAdapter.log(...args),
-    sceneId: bootstrapData.sceneId,
-    eventState,
-    batchEvents,
-    EngineAPI,
-    onEventFunctions,
-    onStartFunctions,
-    onUpdateFunctions
-  })
-
-  // create the context for the scene
-  const runtimeExecutionContext = prepareSandboxContext({
-    dcl,
-    canUseFetch,
-    canUseWebsocket,
-    log: dcl.log,
-    previewMode: isPreview || unsafeAllowed
-  })
-
-  if (bootstrapData.useFPSThrottling === true) {
-    setupFpsThrottling(dcl, fullData.scene.parcels.map(parseParcelPosition), (newValue) => {
-      updateIntervalMs = newValue
+  try {
+    const sourceCode = await codeRequest.text()
+
+    const dcl = createDecentralandInterface({
+      clientPort,
+      onError: (err: Error) => devToolsAdapter.error(err),
+      onLog: (...args: any) => devToolsAdapter.log(...args),
+      sceneId: bootstrapData.sceneId,
+      eventState,
+      batchEvents,
+      EngineAPI,
+      onEventFunctions,
+      onStartFunctions,
+      onUpdateFunctions
     })
-  }
 
-  // run the code of the scene
-  await customEval(sourceCode, runtimeExecutionContext)
+    // create the context for the scene
+    const runtimeExecutionContext = prepareSandboxContext({
+      dcl,
+      canUseFetch,
+      canUseWebsocket,
+      log: dcl.log,
+      previewMode: isPreview || unsafeAllowed
+    })
+
+    if (bootstrapData.useFPSThrottling === true) {
+      setupFpsThrottling(dcl, fullData.scene.parcels.map(parseParcelPosition), (newValue) => {
+        updateIntervalMs = newValue
+      })
+    }
+
+    // run the code of the scene
+    await customEval(sourceCode, runtimeExecutionContext)
+  } catch (err) {
+    await EngineAPI.sendBatch({ actions: [initMessagesFinished()] })
+
+    devToolsAdapter.error(new Error(`SceneRuntime: Error while evaluating the scene ${workerName}`))
+    await sleep(100)
+
+    throw err
```

Comment:
can we eliminate this error since we are already catching it ?

Reply:
If we remove the throw, it fails silently in prod. The sleep is because the devToolsAdapter isn't an async function, if we close the worker the data can be not sent.

Reply:
`sleep` comment added